core/src/main/java/org/mskcc/cbio/portal/servlet/CnaJSON.java

```diff
@@ -167,9 +187,14 @@ private void processGetSegmentsRequest(HttpServletRequest request,
         List<Integer> internalSampleIds = null;
         
         try {
-            int studyId = DaoCancerStudy.getCancerStudyByStableId(cancerStudyId).getInternalId();
-            internalSampleIds = InternalIdUtil.getInternalSampleIds(studyId, Arrays.asList(sampleIds));
-            segs = DaoCopyNumberSegment.getSegmentForSamples(internalSampleIds, studyId);
+        	if(cancerStudyId != null) {
+        		 CancerStudy cancerStudy = DaoCancerStudy
+        	                .getCancerStudyByStableId(cancerStudyId);
+        		if(cancerStudy != null && accessControl.isAccessibleCancerStudy(cancerStudy.getCancerStudyStableId()).size() != 0) {
```

Comment:
You've done this in a few files.  Its minor, but consistency across all files/calls would be nice...thanks!